wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WirePlugin.kt

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (C) 2019 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire.gradle
+
+import com.squareup.wire.gradle.WireExtension.JavaTarget
+import com.squareup.wire.schema.Target
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.file.SourceDirectorySet
+import org.gradle.api.internal.file.FileOrUriNotationConverter
+import org.gradle.api.internal.file.SourceDirectorySetFactory
+import org.gradle.api.plugins.JavaBasePlugin
+import org.gradle.api.tasks.SourceSetContainer
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.api.tasks.compile.JavaCompile
+import org.jetbrains.kotlin.gradle.plugin.KotlinBasePluginWrapper
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import java.io.File
+import java.net.URI
+import javax.inject.Inject
+
+class WirePlugin @Inject constructor(
+  private val sourceDirectorySetFactory: SourceDirectorySetFactory
+) : Plugin<Project> {
+  private var kotlin = false
+  private var java = false
+  override fun apply(project: Project) {
+    val logger = project.logger
+
+    val extension = project.extensions.create(
+        "wire", WireExtension::class.java, project, sourceDirectorySetFactory
+    )
+
+    project.plugins.all {
+      logger.debug("plugin: $it")
+      when (it) {
+        is JavaBasePlugin -> {
+          java = true
+        }
+        is KotlinBasePluginWrapper -> {
+          kotlin = true
+        }
+      }
+    }
+
+    project.afterEvaluate {
+      if (logger.isDebugEnabled) {
+        val ssc = it.property("sourceSets") as SourceSetContainer
+        ssc.forEach {
+          logger.debug("source set: ${it.name}")
+        }
+      }
+
+      when {
+        kotlin || java -> applyJvm(it, extension)
+        else ->
+          throw IllegalArgumentException(
+              "The Wire Gradle plugin requires either the Java or Kotlin plugin to be applied prior to its being applied."
```

Comment:
nit: Is there a better way to phrase this? specifically `requires ... to be applied prior to its being applied`

Reply:
"Either the Java or Kotlin plugin must be applied before the Wire Gradle plugin."?

Reply:
Sounds good!